tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/NativeDesktop/NativeDesktopOperatingSystemsControllerTests.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NHSD.BuyingCatalogue.Solutions.API.Controllers.NativeDesktop;
+using NHSD.BuyingCatalogue.Solutions.API.ViewModels.NativeDesktop;
+using NHSD.BuyingCatalogue.Solutions.Application.Commands.NativeDesktop.UpdateSolutionNativeDesktopOperatingSystems;
+using NHSD.BuyingCatalogue.Solutions.Application.Commands.Validation;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Solutions.API.UnitTests.NativeDesktop
+{
+    [TestFixture]
+    public sealed class NativeDesktopOperatingSystemsControllerTests
+    {
+        private Mock<IMediator> _mockMediator;
+
+        private NativeDesktopOperatingSystemsController _desktopOperatingSystemsController;
+
+        private const string SolutionId = "Sln1";
+
+        [SetUp]
+        public void Setup()
+        {
+            _mockMediator = new Mock<IMediator>();
+            _desktopOperatingSystemsController = new NativeDesktopOperatingSystemsController(_mockMediator.Object);
+        }
+        [TestCase("d")]
+        [TestCase("all sorts of characters, up to 1000, but no more.")]
```

Comment:
these test cases aren't testing anything.

The way the controller tests work, it doesn't matter what gets passed down into mediatr, as the controller isn't doing any validation itself.

You can get away with a single test for each case, as you're mocking out the ISimpleResult. If it's mocked to Valid it should succeed, if it's mocked to Invalid it should fail. The nuances of what causes Valid/Invalid aren't these tests concern.

Reply:
Good point